pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -781,12 +776,191 @@ podCopyPhase:
+		return false
+	}
+
+	return true
+}
+
+// needToRunClusterScopedPlaybook confirms if the cluster scoped playbooks to be run
+func needToRunClusterScopedPlaybook(afwPipeline *AppInstallPipeline) bool {
+	if !isPendingClusterScopeWork(afwPipeline) {
+		return false
+	}
+
+	// Its already time to yield the current reconcile
+	if afwPipeline.afwEntryTime+maxRunTimeBeforeAttemptingYield < time.Now().Unix() {
+		return false
+	}
+
+	return true
+}
+
+// tryAppPkgCleanupFromOperatorPod tries to change the app install status, also cleans the app pkg from Operator Pod
+func tryAppPkgCleanupFromOperatorPod(installWorker *PipelineWorker) {
+	scopedLog := log.WithName("tryAppPkgCleanupFromOperatorPod")
+	if installWorker.sts == nil {
+		scopedLog.Error(nil, "sts is missing", "cr", installWorker.cr.GetName(), "kind", installWorker.cr.GroupVersionKind().Kind, "app pkg", installWorker.appDeployInfo.AppName)
+		deleteAppPkgFromOperator(installWorker)
+		return
+	}
+
+	if *installWorker.sts.Spec.Replicas > 1 {
+		if isAppInstallationCompleteOnAllReplicas(installWorker.appDeployInfo.AuxPhaseInfo) {
+			scopedLog.Info("app pkg installed on all the pods", "app pkg", installWorker.appDeployInfo.AppName)
+			installWorker.appDeployInfo.PhaseInfo.Phase = enterpriseApi.PhaseInstall
+			installWorker.appDeployInfo.PhaseInfo.Status = enterpriseApi.AppPkgInstallComplete
+
+			//For now, set the deploy status as complete. Eventually, we can phase it out
+			installWorker.appDeployInfo.DeployStatus = enterpriseApi.DeployStatusComplete
+			deleteAppPkgFromOperator(installWorker)
+		}
+	} else {
+		deleteAppPkgFromOperator(installWorker)
+	}
+}
+
+// installWorkerHandler fetches and runs the install workers
+func (pplnPhase *PipelinePhase) installWorkerHandler(ppln *AppInstallPipeline, handlerWaiter *sync.WaitGroup, installTracker []chan struct{}) {
+	scopedLog := log.WithName("installWorkerHandler")
+	defer handlerWaiter.Done()
+
+installHandler:
+	for {
+		select {
+		case installWorker, channelOpen := <-pplnPhase.msgChannel:
+			if !channelOpen {
+				// Channel is closed, so, do not handle any more workers
+				scopedLog.Info("worker channel closed")
+				break installHandler
+			}
+
+			if installWorker != nil {
+				podExecClient := splutil.GetPodExecClient(installWorker.client, installWorker.cr, installWorker.targetPodName)
+				podID, _ := getOrdinalValFromPodName(installWorker.targetPodName)
+
+				ctxt := getLocalScopePlaybookContext(installWorker, installTracker[podID], podExecClient)
+				if ctxt != nil {
+					installWorker.waiter.Add(1)
+					go ctxt.runPlaybook()
+				} else {
+					<-installTracker[podID]
+					scopedLog.Error(nil, "unable to get the local scoped context. app name %s", installWorker.appDeployInfo.AppName)
+				}
+			} else {
+				// This should never happen
+				scopedLog.Error(nil, "invalid worker reference")
+			}
+
+		default:
+			time.Sleep(1 * time.Second)
+		}
+
+		time.Sleep(200 * time.Millisecond)
+	}
```

Comment:
Agree, it can be split into two functions. Probably, i will address as part of the refactor Jira, for now will add an item there, considering the time.